Overriding a method just to call the same method from the base class without performing any other actions is useless and misleading. The only time
this is justified is in `sealed` overriding methods, where the effect is to lock in the parent class behavior. This rule ignores overrides
of `Equals` and `GetHashCode`.

## Noncompliant Code Example

<pre>
public override void Method() // Noncompliant
{
  base.Method();
}
</pre>

## Compliant Solution

<pre>
public override void Method()
{
  //do something else
}
</pre>

## Exceptions

If there is an attribute in any level of the overriding chain, then the overridden member is ignored.

<pre>
public class Base
{
  [Required]
  public virtual string Name { get; set; }
}

public class Derived : Base
{
  public override string Name
  {
    get
    {
      return base.Name;
    }
    set
    {
      base.Name = value;
    }
  }
}
</pre>