Property getters should be simple operations that are always safe to call. If exceptions need to be thrown, it is best to convert the property to a
method. 

It is valid to throw exceptions from indexed property getters and from property setters, which are not detected by this rule.

## Noncompliant Code Example

<pre>
public int Foo
{
  get
  {
    throw new Exception(); // Noncompliant
  }
  set
  {
    // do set ...
  }
}
</pre>

## Compliant Solution

<pre>
public void SetFoo(int foo)
{
  // do set ...
}
</pre>