
Merging collapsible `if` statements increases the code's readability.

## Noncompliant Code Example


    If condition1 Then
        If condition2 Then ' Noncompliant
            ' ...
        End If
    End If


## Compliant Solution


    If condition1 AndAlso condition2 Then
        ' ...
    End If

