
Most of the time a block of code is empty when a piece of code is really missing. So such empty block must be either filled or removed.

## Noncompliant Code Example


    ' Empty on purpose or missing piece of code?
    For index As Integer = 1 To 42 ' NonCompliant
    Next


## Exceptions

When a block contains a comment, this block is not considered to be empty.
