
Array designators should always be located on the type for better code readability. Otherwise, developers must look both at the type and the variable name to know whether or not a variable is an array.

## Noncompliant Code Example


    Module Module1
        Sub Main()
            Dim foo() As String ' Noncompliant
        End Sub
    End Module


## Compliant Solution


    Module Module1
        Sub Main()
            Dim foo As String() ' Compliant
        End Sub
    End Module

