
For better readability, do not put more than one statement on a single line.

## Noncompliant Code Example


    Module Module1
        Sub Main()
            Dim a = 0 : Dim b = 0  ' Noncompliant
        End Sub
    End Module


## Compliant Solution


    Module Module1
        Sub Main()
            Dim a = 0              ' Compliant
            Dim b = 0              ' Compliant
        End Sub
    End Module

