
`switch` statements are useful when there are many different cases depending on the value of the same expression.

For just one or two cases however, the code will be more readable with `if` statements.

## Noncompliant Code Example


    Select Case variable
        Case 0
            doSomething()
        Case Else
            doSomethingElse()
    End Select


## Compliant Solution


    If variable = 0 Then
        doSomething()
    Else
        doSomethingElse()
    End If

