
If a `for` loop's condition is false before the first loop iteration, the loop will never be executed. Such loops are almost always<br>bugs, particularly when the initial value and stop conditions are hard-coded.

## Noncompliant Code Example


    for (int i = 10; i &lt; 10; i++)  // Noncompliant
    {
        // ...
    }


## Compliant Solution


    for (int i = 0; i &lt; 10; i++)
    {
        // ...
    }

