
The information that an enumeration type is actually an enumeration or a set of flags should not be duplicated in its name.

## Noncompliant Code Example


    Enum FooFlags ' Noncompliant
        Foo = 1
        Bar = 2
        Baz = 4
    End Enum


## Compliant Solution


    Enum Foo      ' Compliant
        Foo = 1
        Bar = 2
        Baz = 4
    End Enum

