
Calling the `Not` operator twice does nothing: the second invocation undoes the first. Either this is a bug, if the operator was actually meant to be called once, or misleading if done on purpose.

## Noncompliant Code Example


    Dim b As Boolean = False
    Dim c As Boolean = Not Not b 'Noncompliant


## Compliant Solution


    Dim b As Boolean = False
    Dim c As Boolean = b 'Compliant

