
The size of a collection and the length of an array are always greater than or equal to zero. So testing that a size or length is greater than or equal to zero doesn't make sense, since the result is always `true`. Similarly testing that it is less than zero will always return `false`. Perhaps the intent was to check the non-emptiness of the collection or array instead.

## Noncompliant Code Example


    If Collection.Count >= 0 Then ...
    
    If Enumerable.Count < 0 Then ...
    
    If array.Length >= 0 Then ...
    
    Dim result As Boolean = Array.Length >= 0


## Compliant Solution


    If list.Count = 0 Then ...
    If array.Length >= 42 Then ...

