
It is highly suspicious when a value is saved for a key or index and then unconditionally overwritten. Such replacements are likely errors.

## Noncompliant Code Example


    towns.Item(x) = "London"
    towns.Item(x) = "Chicago";  // Noncompliant

